### PR TITLE
Make an OCR function.

### PR DESCRIPTION
**Pull request by uhuntu**
Hi there, this PR is an interesting idea to OCR the png/jpg files.

### CODE REVIEW

**Comment by uhuntu** [comment]
Yes, I re-do it again, please help to review, thanks.

**Comment by picman** [comment]
As the final step, fix the errors in Rubocop tests and we can merge it :-)

**Comment by picman** [comment]
I was thinking for a while about it and found a better approach, I think. We can add a filter to `omindex` command to parse image file directly. E.g.:
_extra/xapian_indexer.rb_

```ruby
...
OMINDEX = "/usr/bin/omindex --filter=image/png:'tesseract %f -'"
...
 # unless File.exist?(OMINDEX)
 #   my_log "#{OMINDEX} does not exist, exiting...", true
 #   exit 1
 # end
...
```
PNG files are then indexed too.

An output on my computer. I have just one PNG with "Karel" in the image.

image

So, I suggest following this approach because:
1. No extra cron job for converting images needed.
2. No duplicite txt images.
3. Support for whatever format, not just PNG or JPG.
4. No changes in the code needed. (I'd just mention this feature in the README and remove the condition for testing the existance of omindex command.

**Comment by uhuntu** [comment]
> ```ruby
> omindex 
> ```

Looks awesome, it is more elegant!

**Comment by uhuntu** [comment]
> I was thinking for a while about it and found a better approach, I think. We can add a filter to `omindex` command to parse image files directly. E.g.: _extra/xapian_indexer.rb_
> 
> ```ruby
> ...
> OMINDEX = "/usr/bin/omindex --filter=image/png:'tesseract %f -'"
> ...
>  # unless File.exist?(OMINDEX)
>  #   my_log "#{OMINDEX} does not exist, exiting...", true
>  #   exit 1
>  # end
> ...
> ```
> 
> PNG files are then indexed too.
> 
> An output on my computer. I have just one PNG with "Karel" in the image.
> 
> image
> 
> So, I suggest following this approach because:
> 
> 1. No extra cron job for converting images needed.
> 2. No duplicite txt images.
> 3. Support for whatever format, not just PNG or JPG.
> 4. No changes in the code needed. (I'd just mention this feature in the README and remove the condition for testing the existance of omindex command.

Looks awesome, it is more elegant!

**Comment by uhuntu** [comment]
Looks awesome, it is more elegant!

**Comment by picman** [comment]
I'm sorry for not accepting your pull request. I've just extended the documentation in #135. Thank you for your effort and contribution to the plugin.

File: extra/xapian_indexer.rb [hunk 1]
```diff
@@ -477,14 +479,31 @@ def find_project(prt)
...
+
 # Indexing files
 unless onlyrepos
   unless File.exist?(OMINDEX)
     my_log "#{OMINDEX} does not exist, exiting...", true
     exit 1
   end
   stem_langs.each do |lang|
-    filespath = File.join(REDMINE_ROOT, FILES)
+    filespath = Redmine::Configuration['attachments_storage_path'] || File.join(REDMINE_ROOT, FILES)
```
Comment (uhuntu) [inline extra/xapian_indexer.rb:506]: @picman I am concerning that if this line can be merged ?

**Pull request closed without merging**